src/CommunityToolkit/Xamarin.CommunityToolkit/Views/DrawingView/Service/DrawingViewService.android.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Android.Graphics;
+using Java.Lang;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+using Color = Xamarin.Forms.Color;
+using Math = System.Math;
+using Point = Xamarin.Forms.Point;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	static class DrawingViewService
+	{
+		/// <summary>
+		/// Get image stream from points
+		/// </summary>
+		/// <param name="points">Drawing points</param>
+		/// <param name="imageSize">Image size</param>
+		/// <param name="lineWidth">Line Width</param>
+		/// <param name="strokeColor">Line color</param>
+		/// <param name="backgroundColor">Image background color</param>
+		/// <returns>Image stream</returns>
+		public static Stream GetImageStream(IList<Point> points,
+			Size imageSize,
+			float lineWidth,
+			Color strokeColor,
+			Color backgroundColor)
+		{
+			if (points == null || points.Count < 2)
+				return Stream.Null;
+
+			var image = GetImageInternal(points, lineWidth, strokeColor, backgroundColor);
+			if (image == null)
+				return Stream.Null;
+
+			using var resizedImage = MaxResizeImage(image, (float)imageSize.Width, (float)imageSize.Height);
+			var stream = new MemoryStream();
+			var compressResult = resizedImage.Compress(Bitmap.CompressFormat.Jpeg, 100, stream);
+
+			resizedImage.Recycle();
+
+			if (!compressResult)
+				return Stream.Null;
+
+			stream.Position = 0;
+			return stream;
+		}
+
+		static Bitmap GetImageInternal(IList<Point> points,
+			float lineWidth,
+			Color strokeColor,
+			Color backgroundColor)
+		{
+			var minPointX = points.Min(p => p.X);
+			var minPointY = points.Min(p => p.Y);
+			var drawingWidth = points.Max(p => p.X) - minPointX;
+			var drawingHeight = points.Max(p => p.Y) - minPointY;
+			const int minSize = 1;
+			if (drawingWidth < minSize || drawingHeight < minSize)
+				throw new Exception($"The image size should be at least {minSize} x {minSize}.");
```

Comment:
@VladislavAntonyuk  I faced this exception and I didn't find any way to intercept it, since it's inside the view. Do we need to throw this exception? Can we just log this message and return?

Reply:
It was my first attempt to understand how nullable works. The original code returned null. It would be better to return null in this case (it is handled by the code above and returns Stream.Null) and set return type as Bitmap?

Reply:
Not sure TBH, but we can give it a try :D

Reply:
I can create a PR if it is necessary

Reply:
If you want, otherwise I'll (try to) do it by tomorrow

Reply:
https://github.com/xamarin/XamarinCommunityToolkit/pull/1339